### PR TITLE
mgr/progress: Add integration to pybind/mgr/tox.ini

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## mgr/progress: Add test_progress.py to tox

`test_progress.py` was previously not integrated to `make check`. As we now have a tox.ini, we can trivially integrate it.

## mgr/progress: Add module to static type checking
(Taken from #26769 )

Some types in this module are not obvious. Half a year ago, adding those type hints really helped me to understand what is going on in this module. As we now have a tox integration, let's make use of it!

--- 
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/test/mgr/test_progress.py [hunk 1]
```diff
@@ -9,6 +9,8 @@
 sys.path.insert(0, "../../pybind/mgr")
 from progress import module
 
+assert False  # who is running this test?
```
Comment (sebastian-philipp) [inline src/test/mgr/test_progress.py]: despite this assert, `make check` passed.

File: src/pybind/mgr/progress/module.py [hunk 2]
```diff
@@ -671,9 +707,9 @@ def _handle_ls(self):
             if len(self._completed_events):
                 # TODO: limit number of completed events to show
                 out += "\n"
-                for ev in self._completed_events:
-                    out += "[{0}]: {1}\n".format("Complete" if not ev.failed else "Failed",
-                                                 ev.twoline_progress())
+                for ghost_ev in self._completed_events:
```
Comment (tchaikov) [inline src/pybind/mgr/progress/module.py]: why shall we encode the type into the variable name?

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-02-03_11:59:37-rados-wip-kefu-testing-2020-02-03-1702-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-02-05_03:04:11-rados-wip-kefu-testing-2020-02-03-1702-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-02-05_09:40:13-rados-wip-kefu-testing-2020-02-03-1702-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
https://tracker.ceph.com/issues/43977

**Comment by tchaikov** [comment]
@sebastian-philipp it's not related. and this failure is exactly the "failed test" i was referencing in https://github.com/ceph/ceph/pull/32881#issuecomment-582391464 .

also could please close https://tracker.ceph.com/issues/43977 ? it's already addressed in the latest change in #32881. and it does not exist in master.

**Comment by tchaikov** [comment]
> I haven't found any "progress" related issue in the runs, do you agree, @tchaikov ?

agreed. that's why i didn't nack this PR. i just don't understand the s/ev/ghost_ev/ change.

File: src/pybind/mgr/progress/module.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/progress/module.py]: https://mypy.readthedocs.io/en/stable/common_issues.html#redefinitions-with-incompatible-types 

`ev` was first declared in line 688.

**Pull request merged**